packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -12,24 +12,187 @@
   options;
   checkStatusPassing({ status_value: 200 } as ResponseStatusCode, 200);
 
+  const browser = await puppeteer.launch({ headless: 'new' });
+  const page = await browser.newPage();
+  await checkLink(
+    page,
+    {} as LinkIntermediate,
+    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions
+  );
+  // END - to resolve warnings  while under development
+
   // options object modified directly
   setDefaultOptions(options);
 
+  // PSEUDOCODE
+
+  // create puppeteer.Browser
+  // create puppeteer.Page & navigate to origin_url, w/ origin specific settings
+
+  // scrape origin_url for all links
+  // (shuffle links if necessary and) truncate at link_limit
+
+  // create new page to be used for all scraped links
+  // navigate to each link - LOOP:
+  //          each call to `checkLinks(...)` will return a `SyntheticLinkResult`
+  //          Object added to an array of `followed_links`
+
+  // returned a SyntheticResult with `options`, `followed_links` &
+  // runtimeMetadata
   return {} as SyntheticResult;
 }
+
+async function checkLink(
+  page: Page,
+  link: LinkIntermediate,
+  options: BrokenLinksResultV1_BrokenLinkCheckerOptions
+): Promise<BrokenLinksResultV1_SyntheticLinkResult> {
+  // START - to resolve warnings while under development
+  page;
+  link;
+  options;
+  // END - to resolve warnings  while under development
+
+  // PSEUDOCODE
+
+  // determine expected_status_code—determined here rather than `navigate(...)`
+  // since it will be used later when creating
+  // `BrokenLinksResultV1_SyntheticLinkResult` Object
+
+  // call `navigate(...)`
+  /**
+   * const {
+   *    response: responseOrError,
+   *    passed,
+   *    testing_only,
+   *    link_start_time,
+   *    link_end_time,
+   * } = await navigate(page, link, expected_status_code, options);
+   */
+
+  // Error handling based on `responseOrError` & passed:
+  //    if      `responseOrError` is type `Error` set `error_type` and
+  //            `error_message` in `SyntheticLinkResult` accordingly
+  //    else if `passed === false`. Set error `SYNTHETIC_INCORRECT_STATUS_CODE`
+
+  // return `SynheticLinkResult` with all calculated information
+  return {} as BrokenLinksResultV1_SyntheticLinkResult;
+}
+
+/**
+ * Navigates to a target URL with retries and timeout handling.
+ *
+ * @param page - The Puppeteer Page instance.
+ * @param link - The LinkIntermediate containing the target URL.
+ * @param expected_status_code - The expected HTTP status code.
+ * @param options - The options for navigation and retries.
+ * @returns Information about navigation attempt:
+ *   - `response`: HTTP response or error if navigation fails, or null.
+ *   - `passed`: Boolean indicating if navigation passed per status code.
+ *   - `retriesRemaining`: Remaining retries after attempt. (for testing)
+ *   - `link_start_time`: Start time of navigation attempt.
+ *   - `link_end_time`: End time of navigation attempt.
+ */
+export async function navigate(
```

Comment:
In my opinion this SDK is more or less supposed to act as an API and perhaps the best way to avoid customers using  `navigate` is just to not export it in index.ts. If users forked the code then they could call the function but if we specificy what to export in index.ts then that would be one work around.

Worth noting that in order to unit test a function, it has to be exported. Possible workarounds:
https://stackoverflow.com/questions/54116070/how-can-i-unit-test-non-exported-functions.

Another option would be to have some structure like this:
- broken_links.ts has `runBrokenLinks(...)` and any exported interfaces
- broken_links_helpers.ts has heavy-lifting function like `navigate`, `checkLink`, ...
- link_utils.ts has small function like creating the SyntheticResult, smaller helpers.

We would only export broken_links.ts so it would be ok if we keep `export` for the util and helper files.
(maybe lets discuss IRL)

Reply:
going to limit access via package level index.ts